### DISCUSSION TITLE
pimpl in C++ | Bhavith C

### DISCUSSION THREAD

**Discussion by giscus** [bhavithc/bhavithc.com#19] [Q&A]
# pimpl in C++ | Bhavith C

What is pimpl

https://bhavithc.com/posts/pimpl/

**Comment by Udaya2019** [comment]
class Student::Impl this is not clear ; Are you inheriting

**Reply by bhavithc** [reply]
`class Student::Impl` is not inheritance it's actual implementation. It is forward declared on line 10 on `Student.h` `class Impl; //forward declaration`

If you still not yet understand please let me know
